src/cpufreq.c

```diff
@@ -82,9 +154,64 @@ static int cpufreq_read(void) {
     /* convert kHz to Hz */
     v.gauge *= 1000.0;
 
-    cpufreq_submit(i, v);
-  }
+    cpufreq_submit(i, "cpufreq", NULL, v);
+
+    /* Read total transitions for cpu frequency */
+    if (report_total_trans) {
+      snprintf(filename, sizeof(filename),
+               "/sys/devices/system/cpu/cpu%d/cpufreq/stats/total_trans", i);
+      fh = fopen(filename, "r");
+      if (fh == NULL)
+        continue;
+      while (fgets(buffer, sizeof(buffer), fh) != NULL) {
+        if (!sscanf(buffer, "%lli", &t)) {
+          fclose(fh);
+          return 0;
+        }
+        snprintf(buffer, sizeof(buffer), "%lli", t - t_data[i].transitions);
```

Comment:
I think you have to use DERIVE type for metrics.
So, delta will be calcultated inside of Collectd core.

Also, there is standard way of file parsing exists, without `snprintf('%lli')` and such ... .

This needs to be rewritten.
The same is for `time_in_state` metric.

Reply:
Based on this comment I have made a couple of changes. 

transitions type is now COUNTER type and diff is calculated inside collectd core. It also uses the standard file parsing based on comment above.

For time_in_state metric parsing is a little more complex as the file is formatted like so:
2401000 335156
2400000 187794999
2300000 1
2200000 393122
Hence I need to parse this file manually to pull out the metrics. 
As per comment I've changed it to calculate the diff inside collectd core and report the percentage time in each frequency state using standard value_to_rate().